nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/ApiModel/Job.cs

```diff
@@ -109,6 +111,90 @@ public bool IsDependencyIgnored(string dependencyName, string dependencyVersion)
             .Any(r => r.IsSatisfiedBy(parsedDependencyVersion));
         return isIgnored;
     }
+
+    public bool IsUpdatePermitted(Dependency dependency)
+    {
+        if (dependency.Version is null)
+        {
+            // if we don't know the version, there's nothing we can do
+            return false;
+        }
+
+        var version = NuGetVersion.Parse(dependency.Version);
+        var dependencyInfo = RunWorker.GetDependencyInfo(this, dependency);
+        var isVulnerable = dependencyInfo.Vulnerabilities.Any(v => v.IsVulnerable(version));
+        var allowed = AllowedUpdates.Any(allowedUpdate =>
+        {
```

Comment:
Nitpick: can we put this into a private local method instead of inline like this? I think it would make things easier to follow.